test/e2e/cluster/add_remove_cluster.go

```diff
@@ -0,0 +1,41 @@
+package cluster
+
+import (
+	"fmt"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/submariner-io/submariner/test/e2e/dataplane"
+	"github.com/submariner-io/submariner/test/e2e/framework"
+)
+
+var _ = Describe("Test to check adding/removing a new cluster to existing setup", func() {
+	f := framework.NewDefaultFramework("add-remove-cluster")
+
+	It("Should be able to add and remove 3rd cluster", func() {
+		By(fmt.Sprintf("Verifying that initially pods from the other clusters can not communicate to service/pod in cluster 1"))
+
+		dataplane.RunNoConnectivityTest(f, false, framework.NonGatewayNode, framework.NonGatewayNode, framework.ClusterA, framework.ClusterC)
+		dataplane.RunNoConnectivityTest(f, true, framework.GatewayNode, framework.NonGatewayNode, framework.ClusterB, framework.ClusterC)
+
+		By(fmt.Sprintf("Adding a new cluster to the setup by setting gateway label to one of the non gateway nodes on cluster 1"))
+		nonGatewayNode := f.FindNodesByGatewayLabel(framework.ClusterC, false)[0].Name
```

Comment:
Sanity check - verify there is at least one gateway node to avoid a panic.

Reply:
I think there would be since we are actually labelling a node just few lines back.

Reply:
I meant non-gateway, ie check that the returned `Node` array is non-empty before dereferencing to avoid a panic even though It *shouldn't* happen.

Reply:
Done